dynamic/dynamic_message_test.go

```diff
@@ -1952,6 +1953,48 @@ func TestMergeFrom(t *testing.T) {
 	// TODO
 }
 
+func TestBuiltMessage(t *testing.T) {
+	md, err := builder.NewMessage("Person").
+		AddField(builder.NewField("name", builder.FieldTypeString())).
+		Build()
+
+	if err != nil {
+		t.Fatalf("error building Person: %v", err)
+	}
+
+	msg := NewMessage(md)
+	msg.SetFieldByNumber(1, "Cailin")
+
+	_, err = desc.LoadMessageDescriptorForMessage(msg)
```

Comment:
Instead of just verifying this doesn't fail, I think it would be better to directly call `msg.Descriptor` to get the dynamic message's descriptor response. And then compare that to `(*testprotos.TestMessage)(nil).Descriptor()`, to verify that the path matches.

Trying to verify that the bytes match will be brittle (since marshaled protos are not necessarily deterministic and GZIP output can vary from one release of Go to the next), so you'd instead need to use `internal.DecodeFileDescriptor` which will decompress the bytes and unmarshal them into a `*FileDescriptorProto`. Then you can use `proto.Equal` to make sure the resulting file descriptors from both the dynamic message and the generated message are the same.

Reply:
BTW, do you `testutil` helpers for the checks:

```go
testutil.Eq(t, expectedPath, actualPath, eqpm, "descriptor paths are not the same")

testutil.Ceq(t, expectedFd, actualFd, eqpm, "file descriptors are not the same")
```

Reply:
Sorry if this comment -- referring to `(*testprotos.TestMessage)(nil)` -- was confusing. I had actually written another comment that indicated using generated types in the `testprotos` package instead of creating types with the `builder` package. I guess I accidentally lost that comment though. Anyhow, changes look fine.